usr.sbin/daemon/daemon.c

```diff
@@ -613,19 +588,18 @@ restrict_process(const char *user)
  * Return value of false is assumed to mean EOF or error, and true indicates to
```

Comment:
is there anything special to do in case of error?
if so, you might create an Enum with SUCCESS, ERROR, EOF

Reply:
A nice improvement for the future. I was avoiding `listen_child` function for now, it still exists in its initial form (except for int->bool return type change)